MultiSource/Benchmarks/Ptrdist/anagram/anagram.c

```diff
@@ -440,14 +526,30 @@ void BuildWord(char * pchWord) {
 /* AddWords -- build the list of candidates */
 void
 AddWords(void) {
-    char * pch = pchDictionary;     /* walk through the dictionary */
+    // CHECKEDC : declared bounds at variable declaration
+    // dictionary max length = MAXWORDS
+    _Array_ptr<char> pch : bounds(pchDictionary, pchDictionary+MAXWORDS) = pchDictionary;     /* walk through the dictionary */
```

Comment:
I think you will need to have local variables that hold the bounds for pch.   The problem is that at function calls, we have to assume that pchDictionary and its length might change.

For example:
```
_Array_ptr<char> pchLowerBounds = pchDictionary;
_Array_ptr<char> pchUpperBounds = pchDictionary + pchDictionarySize;
_Array_ptr<char> pch : bounds(pchLowerBounds, pchUpperbounds) = pchDictionary;
```